src/core/backbone_router/local.cpp

```diff
@@ -66,14 +67,18 @@ void Local::SetEnabled(bool aEnable)
     if (aEnable)
     {
         SetState(OT_BACKBONE_ROUTER_STATE_SECONDARY);
+        AddDomainPrefixToNetworkData();
```

Comment:
Looking at this again, should we not enable BBR service if we fail to add necessary items to network data?

Reply:
One BBR Service takes 16B, one Domain Prefix TLV takes 20B, when fails to add domain prefix, it might be still possible to add BBR service successfully? Should Multicast across Thread Networks be independent regardless of whether or not there is DUA feature and whether or not DUA is enabled successfully?

In general I don't think there is big chance for the local server data be full,  otherwise, bigger chance for the leader network data to be much more insufficient.

The size limitation of Thread Network Data should be a general issue when there are too much servers - like > 12 (=254/20) different prefixes configuration, which may be assumed to not happen generally?

Thoughts?

Reply:
How about adding log messages if adding to network data fails?

Reply:
Also, if we not enable BBR service when we failed to add domain prefix to network data, then we are not allowing empty domain prefix.

Reply:
> Also, if we not enable BBR service when we failed to add domain prefix to network data, then we are not allowing empty domain prefix.

@simonlingoogle, I didn't mean any error. If the error is `OT_ERROR_NO_BUFS`, that could be an issue for the leader.